runelite-client/src/main/java/net/runelite/client/plugins/duplicatechat/DuplicateChatPlugin.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (c) 2018, Magic fTail
+ * Copyright (c) 2019, osrs-music-map <osrs-music-map@users.noreply.github.com>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.duplicatechat;
+
+import com.google.common.collect.HashMultiset;
+import com.google.common.collect.Multiset;
+import com.google.inject.Provides;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.ChatMessageType;
+import net.runelite.api.Client;
+import net.runelite.api.MessageNode;
+import net.runelite.api.events.ChatMessage;
+import net.runelite.api.events.ScriptCallbackEvent;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.eventbus.Subscribe;
+import net.runelite.client.events.ConfigChanged;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.util.ColorUtil;
+
+import javax.inject.Inject;
+import java.util.HashMap;
+import java.util.Map;
+
+@PluginDescriptor(
+		name = "Duplicate Chat Filter",
+		description = "Collapse duplicate game chats into a single message",
+		enabledByDefault = false
+)
+@Slf4j
+public class DuplicateChatPlugin extends Plugin
+{
+
+	private static final String COUNT_FORMAT = " (%d)";
+	private Multiset<String> chatCounts;
+	private Map<String, Integer> chatIndex;
```

Comment:
Can't these both be replaced by a single `MultiMap<String, Integer>`?

Reply:
No, I don't think so. The chatCounts keeps track of the amount of duplicate chat messages with the key being the message, and the chatIndex keeps track of the newest index of the message in question, so that the chatfilter can tell if it should be blocked or not

Reply:
Isn't that achieved by checking the following in a `ListMultiMap`?

- Number of duplicated chat messages: `multiMap.get(str).size()`
- Newest index of message in question: `Iterables.getLast(multiMap.get(str))`

Reply:
I see what you are saying now, that makes a lot of sense. I made the change to use a multimap, it definitely looks cleaner now. Thank you for the suggestion!